docs/en/operations/settings/constraints-on-settings.md

```diff
@@ -68,6 +68,32 @@ Code: 452, e.displayText() = DB::Exception: Setting max_memory_usage should not
 Code: 452, e.displayText() = DB::Exception: Setting force_index_by_date should not be changed.
 ```
 
+## Constraints in read-only mode {#settings-readonly-constraints}
+Read-only mode is enabled by `readonly` setting:
+- `readonly=0`: No read-only restrictions.
+- `readonly=1`: Only read queries are allowed and settings cannot be changes unless explicitly allowed.
+- `readonly=2`: Only read queries are allowed, but settings can be changed, except for `readonly` setting itself.
```

Comment:
It looks confusing because 0 is the least restrictive and 1 is the most, so 2 is between them. But I see that we need to keep 0 and 1 for compatibility.